src/pages/workspace/WorkspaceInviteMessagePage.js

```diff
@@ -0,0 +1,211 @@
+
+        this.onSubmit = this.onSubmit.bind(this);
+        this.validate = this.validate.bind(this);
+        this.openPrivacyURL = this.openPrivacyURL.bind(this);
+        this.state = {
+            welcomeNote: this.getWelcomeNote(),
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            !(prevProps.preferredLocale !== this.props.preferredLocale
+            && this.state.welcomeNote === Localize.translate(prevProps.preferredLocale, 'workspace.inviteMessage.welcomeNote', {workspaceName: this.props.policy.name})
+            )
+        ) {
+            return;
+        }
+        this.setState({welcomeNote: this.getWelcomeNote()});
+    }
+
+    onSubmit() {
+        const filteredLogins = _.uniq(_.compact(_.map(this.props.invitedMembersDraft, login => login.toLowerCase().trim())));
+        Policy.addMembersToWorkspace(filteredLogins, this.state.welcomeNote || this.getWelcomeNote(), this.props.route.params.policyID);
+        Navigation.navigate(ROUTES.getWorkspaceMembersRoute(this.props.route.params.policyID));
+    }
+
+    getAvatars() {
+        const filteredPersonalDetails = _.pick(this.props.personalDetails, this.props.invitedMembersDraft);
+        return _.map(filteredPersonalDetails, personalDetail => ({
+            source: ReportUtils.getAvatar(personalDetail.avatar, personalDetail.login),
+            type: CONST.ICON_TYPE_AVATAR,
+            name: personalDetail.login,
+        }));
+    }
+
+    getWelcomeNote() {
+        return this.props.translate('workspace.inviteMessage.welcomeNote', {
+            workspaceName: this.props.policy.name,
+        });
+    }
+
+    getAvatarTooltips() {
+        const filteredPersonalDetails = _.pick(this.props.personalDetails, this.props.invitedMembersDraft);
+        return _.map(filteredPersonalDetails, personalDetail => Str.removeSMSDomain(personalDetail.login));
+    }
+
+    openPrivacyURL(e) {
+        e.preventDefault();
+        Link.openExternalLink(CONST.PRIVACY_URL);
+    }
+
+    /**
+     * @returns {Boolean}
+     */
+    validate() {
+        // No validation required as the invite message is optional
+        return {};
+    }
+
+    render() {
+        const policyName = lodashGet(this.props.policy, 'name');
+
+        return (
+            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+                <HeaderWithCloseButton
+                    title={this.props.translate('workspace.inviteMessage.inviteMessageTitle')}
+                    subtitle={policyName}
+                    shouldShowGetAssistanceButton
+                    guidesCallTaskID={CONST.GUIDES_CALL_TASK_IDS.WORKSPACE_MEMBERS}
+                    shouldShowBackButton
+                    onCloseButtonPress={() => Navigation.dismissModal()}
+                    onBackButtonPress={() => Navigation.goBack()}
+                />
+                <Form
+                    style={[styles.flexGrow1, styles.ph5]}
+                    formID={ONYXKEYS.FORMS.LEGAL_NAME_FORM}
```

Comment:
@abdulrahuman5196 Why are we using this key instead creating a new one for this form?